### ISSUE TITLE
[feature]: improve usability as a dev tool

### ISSUE THREAD

**Issue by jessfraz** [Azure/acs-engine#2535]
I spin up a lot of clusters for testing and I realize this is not really the use case of acs-engine, but I do think it is a valid one.

I would love a `delete/rm` command, right now i have bash scripts that do this, but if I could stay in the same tool that would be even more awesome.

I would love to be able to pass an ssh-key to `deploy` so i don't have to edit the config in the repo, then make sure not to git commit that. Or then I don't have to use my bash script to add the generated key into my keychain and scp the generated key onto the master node so that i can ssh into the others blah blah blah.

I would love a list command because then I could pipe list to my dream `delete/rm` command :) i get that _output is not like "a thing" but we could make it so.

**Comment by jessfraz** [comment]
Oh and list should show the master ip in a table like so, so i dont have to look it up in the portal.

```
name                                               ip
jessfraz-clear-containers-{uuid}      1.2.3.4
```

**Comment by jessfraz** [comment]
Oh and I'd love a command that takes that deployments kubeconfig and adds it to my local one :)

So four things:

- [ ] delete/rm
- [ ] --ssh-key on deploy
- [ ] list with ip of master
- [ ] command to move kubeconfig locally

**Comment by jessfraz** [comment]
I wrote a wrapper bash script that does all these things: https://github.com/jessfraz/dotfiles/blob/master/bin/acsengine

```
$ acsengine ls
jessfraz-clear-containers-5aba412c 52.168.169.97

$ acsengine kubeconfig jessfraz-clear-containers-5aba412c
Merged kubeconfig for jessfraz-clear-containers-5aba412c to local kubeconfig
Run `kubectl config view` to verify

$ acsengine rm jessfraz-clear-containers-5aba412c
```

**Comment by khenidak** [comment]
`acs-engine create --resource-group {RG} --master-count {N} --agent-count {N} --ssh-key--value {PATH} --kubernetes-version {VER}`

Driven from a a golden apimodel where we have a minimal set of overridable arguments. Everything else is according to this api-model.

*or*

`acs-engine create --base-model {PATH} --resource-group {RG} --master-count {N} --agent-count {N} --ssh-key--value {PATH} --kubernetes-version {VER}` 

basically repeatable painless create experience.

my 2cents.

**Comment by yakman2020** [comment]
acs engine supports both k8s and dcos. Presumably there would be  some parameterisation to reflect that?

**Comment by jackfrancis** [comment]
May I add an item to this aspirational playground?

- Persistent user context config, i.e., `~/.acs-engine/config.json` or somesuch

**Comment by khenidak** [comment]
Let us label this with `help wanted` @jackfrancis
